centrifuge/build.gradle

```diff
@@ -26,6 +26,7 @@ dependencies {
     implementation 'com.squareup.okhttp3:okhttp:4.10.0'
     implementation 'com.google.protobuf:protobuf-javalite:3.21.12'
     implementation 'net.sourceforge.streamsupport:streamsupport-cfuture:1.7.0'
+    compileOnly 'com.google.code.findbugs:jsr305:3.0.2'
```

Comment:
I've [found](https://github.com/square/retrofit/blob/c956f82929cf9f9ecf296a8337107089b8032ca9/retrofit-converters/guava/build.gradle#L7) that you don't need implementation here=)

I think it's totally fine that users should also add this (or analog dependency) to their project for nullability support, or disable nullability checks in some way

Reply:
I tested it with publishToMavenLocal, and it results in annotation completely missing. As if methods and arguments don't have any nullability annotations.

Reply:
It looks like there is no way to avoid those warning in java projects.
But I would personally prefer library to automatically provide the jsr305 dependency. What is the rationale behind using `compileOnly` vs `api` here?

Reply:
There're different sets of nullable annotation that can be used in a project: `javax.annotation.Nullable`/`org.jetbrains.annotations.Nullable`/`androidx.annotation.Nullable`, etc. They all work the same and can be mixed without problems.

So if user choose a set different from ours in his project, after adding the library with the `api` dependency, his suggestions will be cluttered with multiple options.

Also, Kotlin users won't use annotation at all, so they don't need it.

Reply:
I tried to test different cases in java and kotlin, compileOnly works great in both. The only issue is that if java developers will want to fix those warnings, that would need to manually add jsr305 dependency.

Reply:
Yeah, but I assume that if you're java dev, either already use them in your code for the better IDE checks, or disable these checks completely.

In cases where I'm not sure which way is more correct, I prefer to follow the approaches of popular/reliable libraries, that's why I took `compileOnly` from Retrofit.

Reply:
gotcha! I that makes sense. 

Btw, in your example (for guava converter for retrofit), guava library [already includes jsr305](https://mvnrepository.com/artifact/com.google.guava/guava/29.0-jre) as its compile dependency.